### PR TITLE
fix: improve UX of team invite by copy link

### LINKED ISSUE #10906: [CAL-2381] invited by link throws 500
inviting someone to a team via link is currently throwing a 500

<sub>[CAL-2381](https://linear.app/calcom/issue/CAL-2381/invited-by-link-throws-500)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/10906

1) checked user session in getServersideprops to speed up redirecting to /login page if unauthenticated instead of showing unauthorised error on /teams and then redirecting which was very slow.

2) added alert to ask user to "sign in or sign up to see team invite".

https://github.com/calcom/cal.com/assets/53316345/7af9a602-6920-4910-8b01-a7b1af111cf3

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/signup.tsx [hunk 1]
```diff
@@ -72,11 +75,12 @@ export default function Signup({ prepopulateFormValues, token, orgSlug }: Signup
       .then(async () => {
         telemetry.event(telemetryEventTypes.signup, collectPageParameters());
         const verifyOrGettingStarted = flags["email-verification"] ? "auth/verify-email" : "getting-started";
+
         await signIn<"credentials">("credentials", {
           ...data,
           callbackUrl: `${
             searchParams?.get("callbackUrl")
-              ? `${WEBAPP_URL}/${searchParams.get("callbackUrl")}`
+              ? `${searchParams.get("callbackUrl")}`
```
Comment (zomars) [inline apps/web/pages/signup.tsx:83]: This could be tampered to redirect you to unwanted sites isn't it? cc @hariombalhara

**Review by zomars — CHANGES_REQUESTED** [review]
Overall nice improvements @Udit-takkar just left a concern about callbackUrl to address before merging 🙏🏽

File: apps/web/pages/signup.tsx (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/pages/signup.tsx]: should be fine now with getSafeCallbackUrl

**Review by zomars — APPROVED** [review]
Nice. Nice job @Udit-takkar 

Looks good to merge

LGTM

**Pull request merged**